### PR TITLE
Add support for using black to format Python code

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

File: src/esss_fix_format/cli.py [hunk 1]
```diff
@@ -56,6 +58,23 @@ def should_format(filename):
     return False, 'Unknown file type'
 
 
+def find_black_config(files_or_directories) -> Union[Path, None]:
+    """
+    Searches for a valid pyproject.toml file based on the list of files/directories given.
+
+    We find the common ancestor of all files/directories given, then search from there
+    upwards for a "pyproject.toml" file with a "[tool.back]" section.
```
Comment (igortg) [inline src/esss_fix_format/cli.py]: typo "back"

File: src/esss_fix_format/cli.py [hunk 2]
```diff
@@ -56,6 +58,23 @@ def should_format(filename):
     return False, 'Unknown file type'
 
 
+def find_black_config(files_or_directories) -> Union[Path, None]:
```
Comment (tadeu) [inline src/esss_fix_format/cli.py]: `Union[Path, None]` could also be `Optional[Path]`

File: src/esss_fix_format/cli.py [hunk 3]
```diff
@@ -304,9 +352,22 @@ def _main(files_or_directories, check, stdin, commit, format_code):
...
-    changed_files = []
+
+    files = [Path(x) for x in files]
     errors = []
+
+    black_config = find_black_config(files)
+    formatted_by_black = False
+    if black_config:
+        # skip pydevf formatter
+        format_code = None
```
Comment (igortg) [inline src/esss_fix_format/cli.py]: Wouldn't be better to have this variable renamed to `pydevf_format` or something alike?

File: src/esss_fix_format/cli.py [hunk 4]
```diff
@@ -291,7 +311,35 @@ def _process_file(filename, check, format_code):
...
+            click.secho('Checking black...', fg='cyan')
+        else:
+            click.secho('Running black...', fg='cyan')
+        args = ['black']
+        if check:
+            args.append('--check')
+        args.extend(str(x) for x in py_files)
+        status = subprocess.call(args)
+        error = not check and status != 0
+        formatted = check and status == 1
```
Comment (tadeu) [inline src/esss_fix_format/cli.py]: Perhaps `formatted` should be called `should_format`, to be more clear? (From black: `Return code 1 means some files would be reformatted.`)

(also in line 364)

Comment (nicoddemus) [reply src/esss_fix_format/cli.py]: Definitely, not sure what I was thinking...

File: README.rst [hunk 5]
```diff
@@ -119,7 +119,10 @@ Follow this steps to re format an entire project and start using the pre-commit
         $ cd /path/to/repo/root
         $ curl -O https://raw.githubusercontent.com/ESSS/esss_fix_format/master/.clang-format
 
-5. Execute:
+5. If you want to use ``black`` to format Python code, add a ``pyproject.toml`` to the root of
+   your repository; an example can be found above.
+
+6. Execute:
```
Comment (prusse-martin) [inline README.rst]: In the example we could make note that the proper environment must be active:

```
**    .. code-block:: sh
        $ cd /path/to/repo/root
        $ conda activate name_of_project_env  # not necessary if proper environment is active
        $ ff .
```

**Pull request merged**